app/models/miq_template.rb

```diff
@@ -31,8 +31,7 @@ def scan_via_ems?
   end
 
   def self.eligible_for_provisioning
-    types = descendants.select { |d| d.supports?(:provisioning) }.map(&:name)
-    where(:type => types).where.not(:ems_id => nil)
+    supporting(:provisioning).where.not(:ems_id => nil)
```

Comment:
Not for this PR but just noticed this can use the `.active` scope from `VmOrTemplate` (just `where.not(:ems_id => nil)`)